src/types/basic.rs

```diff
@@ -270,22 +272,158 @@ pub struct TransactionIndex {
 
 pub type AggregateSigPairing = id::constants::IpPairing;
 
+/// FIXME: Move higher up in the dependency
+#[derive(SerdeBase16Serialize, Serialize)]
+pub struct BakerAggregationSignKey {
+    pub(crate) sign_key: aggregate_sig::SecretKey<AggregateSigPairing>,
+}
+
+impl BakerAggregationSignKey {
+    pub fn generate<T: Rng>(csprng: &mut T) -> Self {
+        Self {
+            sign_key: aggregate_sig::SecretKey::generate(csprng),
+        }
+    }
+
+    pub fn prove<T: Rng>(
```

Comment:
Document what this function proves since it is part of the public API.

Reply:
I have added a doc comment